tests/test_util.py

```diff
@@ -219,6 +219,22 @@ def test_complex_conditions_expr(self):
         assert complex_column_expr(tuplify(['top3', ['project_id']]), body.copy()) == 'topK(3)(project_id)'
         assert complex_column_expr(tuplify(['top10', ['project_id'], 'baz']), body.copy()) == '(topK(10)(project_id) AS baz)'
 
+        assert complex_column_expr(tuplify(['or', [['a', '=', 1], ['b', '=', 1]]]), body.copy()) == 'or(a = 1, b = 1)'
+        assert complex_column_expr(tuplify(['and', [['a', '=', 1], ['b', '=', 1]]]), body.copy()) == 'and(a = 1, b = 1)'
```

Comment:
it seems like having conditions as function args creates a potential ambiguity here, eg the input:
```
['and', ['foo', ['a', '=', 1]]]
```
could either be interpreted as:
```
and(foo, a = 1)
```
or a nested function
```
and(foo('a', '=', 1))
```

Reply:
It doesn't. There is another bracket you're missing. 

```
['and', ['foo', ['a', '=', 1]]]
```
Should instead be
```
['and', [ ['foo', [ ['a', '=', 1] ]] ]]
```

This is because in the general case a function is:
```
[func [arg0, arg1, ...]]
```
where `argi` can be a condition like `['a', '=', '1']` or `['foo' [['a','=','1']]]`

Reply:
Right, so the problem comes when you have a function where the first argument is a column name, and the second argument is a condition. Eg in a dataset with 2 integer columns, `A` and `B`

To get `and(A, B)` you send `["and", ["A", "B"]]` (per your tests in https://github.com/getsentry/snuba/pull/310)

But if I want to instead check that B is some specific value (not just truthy like above) then I might want something like `and(A, B=2)` which means that the second argument to my `and` function is a condition and it would look something like `["and", ["A", ["B", "=", 2]]]`

But unfortunately the syntax uses "a string followed by a list" to mean a function call, so we would interpret `"A"` as a function name instead of a column name, and `["B", "=", 2]` as an argument list instead of a condition. and the whole thing would parse out as something like `and(A(B, "=", 2))` instead.

Reply:
This is basically why you can't pass an array as an argument to a function currently, because if that array was the 2nd argument (after a string argument), we wouldn't know whether it was a list arg or a nested function call